### ISSUE TITLE
Pre-trained caffe model for FCN score evaluation

### ISSUE THREAD

**Issue by FishYuLi** [phillipi/pix2pix#116]
Hello, could you please tell me the performance of the pre-trained caffe model on the original cityscapes datasets given by  [this code](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/download_fcn8s.sh) ?
I follow the instructions in `readme.mk` to evaluate the label_to_image generator, which means the input of the FCN model is the generated fake image. I wondered how the model downloaded by [this code](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/download_fcn8s.sh) works on the original cityscapes datasets. So I change the parameter `--result_dir` in [./scripts/eval_cityscapes/evaluate.py](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/evaluate.py) to the original `val` set of cityscapes. But the results are very bad(as follows). Is there anything wrong with this model ? Or is this model already been trained on cityscapes dataset?
![segresult](https://user-images.githubusercontent.com/17564896/33012971-5ff73684-ce1d-11e7-9313-d257e0722c59.png)
Thank you.

**Comment by wxh001qq** [comment]
sorry, may i ask you how to run the evaluate.py?  i found there request  _labels_,and there are many fold that could not found in the cityscape dataset that the download.sh provide

**Comment by FishYuLi** [comment]
@wxh001qq 
1. You need to download the original labels of  [cityscape dataset](https://www.cityscapes-dataset.com) (not the given dataset in this code).
2. Change the paths in [here](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/cityscapes.py) to make sure that you can fine everything you need.

**Comment by tinghuiz** [comment]
The pre-trained model does not work well on Cityscapes in the original resolution (1024x2048) as it was trained on 256x256 images that are resized to 1024x2048. The purpose of the resizing was to 1) keep the label maps in the original resolution and 2) avoid the need of changing the standard FCN training code for Cityscapes. To get the 'ground-truth' numbers in the paper, you need to resize the original Cityscapes images to 256x256 before running the evaluation code.

**Comment by tychovdo** [comment]
This issue seems similar a [problem](https://github.com/phillipi/pix2pix/issues/148) that I am having. I tried resizing the images to 256x256, but I still get really bad results from the pretrained caffe model.
@FishYuLi : did you find a solution to this problem?

**Comment by tychovdo** [comment]
Your ‘bad looking’ segmentations do look very similar to the segmentations that I am getting (mostly “road”). How did you resize the images?
I use
‘’’
from skimage.misc import imresize
resized = imresize(image, (256, 256, 3))
‘’’

**Comment by FishYuLi** [comment]
@tychovdo I just remembered that if you want to run fcn score on the original image, you need to first resize the image to 256x256, and resize it to 1024x2048, and then send it to the segmentation network. Just uncomment [this line](https://github.com/phillipi/pix2pix/blob/4b4621dbeb7f22a0335e1f2fc598a8782b6b7240/scripts/eval_cityscapes/evaluate.py#L46).
It has been a long time when I met this problem, so I am not very sure. But you may try.

**Comment by lx7555** [comment]
非常感谢回复。按照你当时的留言，1我目前已经将cityscapes数据集中自带的scripts下载好，我想这样就解决了request "labels"的问题。
2,我已经下载好cityscapes数据集原图(leftImg8bit)和精细标注(gtFine_trainvaltest)。请问如何安排和放置文件夹的位置，能够让评估程序读取到数据集的信息？
3.你说通过将cityscapes数据集压缩成256*256再放大成1024*2048，再将某一行代码取消注释，可以评估cityscapes数据集的原始像素图像。原文说只能评估256*256，请问你取消代码注释后。评估cityscapes数据集原始像素图像效果如何？为什么要先裁剪，后还原？？

**Comment by lkiani** [comment]
> @wxh001qq
> 
> 1. You need to download the original labels of  [cityscape dataset](https://www.cityscapes-dataset.com) (not the given dataset in this code).
> 2. Change the paths in [here](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/cityscapes.py) to make sure that you can find everything you need.

I downloaded the labels.py file and placed in content / pix2pix / scripts / helpers / but I still receive the No module named labels message.
thank you

**Comment by ZhangCZhen** [comment]
@FishYuLi  Hello, my evaluation results are the same as yours. In particular, 
1. If the images are 1024*2048, the mean pixel accuracy = 0.479361, mean class accuracy = 0.087739, mean class IoU = 0.056188.
2. If the images are 256*256, the mean pixel accuracy = 0.793441, mean class accuracy = 0.237466, mean class IoU = 0.1963344.
I evaluated using RGB images from real datasets, and I think the accuracy value should be close to 1. What is the result of your reproduction? What is your reproduction process? Thanks a lot!

**Comment by FishYuLi** [comment]
@ZhangCZhen Please refer to [the previous discussion.](https://github.com/phillipi/pix2pix/issues/116#issuecomment-415251319)
I got the same FCN-scores as stated in Pix2pix[1] Table.1. (0.80, 0.26, 0.21 for per-pixel acc. per-class acc. and class IOU). It won't be 1, becuase it is decided by the performance of the FCN model you used to evaluate FCN-score.


[1] Phillip Isola et al. Image-to-Image Translation with Conditional Adversarial Networks.

**Comment by ZhangCZhen** [comment]
@FishYuLi    Thank you sooooooo much for your help!

**Comment by syz825211943** [comment]
Hello, the link of pre-trained caffe model is unavailable, can you share the model to me? thx! @FishYuLi @ZhangCZhen

**Comment by HuXiaokai12138** [comment]
> Your ‘bad looking’ segmentations do look very similar to the segmentations that I am getting (mostly “road”). How did you resize the images? I use
> 
> ```
> from skimage.misc import imresize
> resized = imresize(image, (256, 256, 3))
> ```

I would like to know how you ended up changing the size of the image and look forward to hearing from you.

**Comment by ALISAXXW** [comment]
您的邮件已收到，我会尽快回复~~~

**Comment by HuXiaokai12138** [comment]
> Your ‘bad looking’ segmentations do look very similar to the segmentations that I am getting (mostly “road”). How did you resize the images? I use
> 
> ```
> from skimage.misc import imresize
> resized = imresize(image, (256, 256, 3))
> ```

@tychovdo I would like to know how you ended up changing the size of the image and look forward to hearing from you.

**Comment by HuXiaokai12138** [comment]
> @FishYuLi Hello, my evaluation results are the same as yours. In particular,
> 
> 1. If the images are 1024*2048, the mean pixel accuracy = 0.479361, mean class accuracy = 0.087739, mean class IoU = 0.056188.
> 2. If the images are 256*256, the mean pixel accuracy = 0.793441, mean class accuracy = 0.237466, mean class IoU = 0.1963344.
>    I evaluated using RGB images from real datasets, and I think the accuracy value should be close to 1. What is the result of your reproduction? What is your reproduction process? Thanks a lot!

I would like to know how you ended up changing the size of the image and look forward to hearing from you.

**Comment by HuXiaokai12138** [comment]
> Hello, could you please tell me the performance of the pre-trained caffe model on the original cityscapes datasets given by [this code](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/download_fcn8s.sh) ? I follow the instructions in `readme.mk` to evaluate the label_to_image generator, which means the input of the FCN model is the generated fake image. I wondered how the model downloaded by [this code](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/download_fcn8s.sh) works on the original cityscapes datasets. So I change the parameter `--result_dir` in [./scripts/eval_cityscapes/evaluate.py](https://github.com/phillipi/pix2pix/blob/master/scripts/eval_cityscapes/evaluate.py) to the original `val` set of cityscapes. But the results are very bad(as follows). Is there anything wrong with this model ? Or is this model already been trained on cityscapes dataset? ![segresult](https://user-images.githubusercontent.com/17564896/33012971-5ff73684-ce1d-11e7-9313-d257e0722c59.png) Thank you.

@FishYuLi I have the same problem, even the numbers are the same as yours This question has been bothering me for a long time, I would love to know how you solved this problem, and I look forward to your reply.

**Comment by HuXiaokai12138** [comment]
> @FishYuLi Hello, my evaluation results are the same as yours. In particular,
> 
> 1. If the images are 1024*2048, the mean pixel accuracy = 0.479361, mean class accuracy = 0.087739, mean class IoU = 0.056188.
> 2. If the images are 256*256, the mean pixel accuracy = 0.793441, mean class accuracy = 0.237466, mean class IoU = 0.1963344.
>    I evaluated using RGB images from real datasets, and I think the accuracy value should be close to 1. What is the result of your reproduction? What is your reproduction process? Thanks a lot!

@ZhangCZhen I would like to ask, did you change the image to 256×256 only the real picture, or did you change the label map as well?Because I got the same result as you before I changed it.
